packages/wrangler/src/versions/api.ts

```diff
@@ -104,28 +108,62 @@ export async function createDeployment(
 	accountId: string,
 	workerName: string,
 	versionTraffic: Map<VersionId, Percentage>,
-	message: string | undefined
+	message: string | undefined,
+	force?: boolean
 ) {
-	const res = await fetchResult(
-		`/accounts/${accountId}/workers/scripts/${workerName}/deployments`,
-		{
-			method: "POST",
-			headers: { "Content-Type": "application/json" },
-			body: JSON.stringify({
-				strategy: "percentage",
-				versions: Array.from(versionTraffic).map(
-					([version_id, percentage]) => ({ version_id, percentage })
-				),
-				annotations: {
-					"workers/message": message,
-				},
-			}),
+	try {
+		return await fetchResult(
+			`/accounts/${accountId}/workers/scripts/${workerName}/deployments${force ? "?force=true" : ""}`,
+			{
+				method: "POST",
+				headers: { "Content-Type": "application/json" },
+				body: JSON.stringify({
+					strategy: "percentage",
+					versions: Array.from(versionTraffic).map(
+						([version_id, percentage]) => ({ version_id, percentage })
+					),
+					annotations: {
+						"workers/message": message,
+					},
+				}),
+			}
+		);
+	} catch (e) {
+		if (
+			e instanceof APIError &&
+			e.code === CANNOT_ROLLBACK_WITH_MODIFIED_SECERT_CODE
+		) {
+			// This is not great but is the best way I could think to handle for now
+			const errorMsg = e.notes[0].text.replace(
+				` [code: ${CANNOT_ROLLBACK_WITH_MODIFIED_SECERT_CODE}]`,
+				""
+			);
+			const targetString = "The following secrets have changed:";
+			const changedSecrets = errorMsg
+				.substring(errorMsg.indexOf(targetString) + targetString.length + 1)
+				.split(", ");
+
+			const confirmed = await confirm(
```

Comment:
what happens if we are not in interactive mode? should we check for `isInteractive()`?

Reply:
It defaults and moves on - added a test for this :)